tests/test_generate.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import shutil
+
+from docuploader import shell
+from google.cloud import storage
+from google.oauth2 import service_account
+import pytest
+
+from docpipeline import generate, tar
+
+
+@pytest.fixture
+def yaml_dir(tmpdir):
+    shutil.copytree("testdata", tmpdir, dirs_exist_ok=True)
+    return tmpdir
+
+
+def test_generate(yaml_dir, tmpdir):
+    test_bucket = os.environ.get("TEST_BUCKET")
+    if not test_bucket:
+        pytest.skip("Must set TEST_BUCKET")
+    return test_bucket
```

Comment:
I think `return` here will end the test.

Reply:
Whoops! Fixed. Good catch.